### PR TITLE
[extractor/trtcocuk] Add extractor

### LINKED ISSUE #2635: trtcocuk.net.tr
### Checklist

- [x] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2022.02.04**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Turkey

### Description

When try to get video from yt-dlp gives error about bad request.

ERROR: [generic] Unable to download webpage: HTTP Error 400: Bad Request

### Verbose log

```shell
da@ffof:~$ yt-dlp -vU https://www.trtcocuk.net.tr/video/kaptan-pengu-ve-arkadaslari-1
[debug] Command-line config: ['-vU', 'https://www.trtcocuk.net.tr/video/kaptan-pengu-ve-arkadaslari-1']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8
[debug] yt-dlp version 2022.02.04 [c1653e9ef] (zip)
[debug] Python version 3.9.7 (CPython 64bit) - Linux-5.15.15-76051515-generic-x86_64-with-glibc2.34
[debug] exe versions: ffmpeg 4.4 (setts), ffprobe 4.4, rtmpdump 2.4
[debug] Optional libraries: Cryptodome, secretstorage, sqlite
[debug] Proxy map: {}
Latest version: 2022.02.04, Current version: 2022.02.04
yt-dlp is up to date (2022.02.04)
[debug] [generic] Extracting URL: https://www.trtcocuk.net.tr/video/kaptan-pengu-ve-arkadaslari-1
[generic] kaptan-pengu-ve-arkadaslari-1: Requesting header
WARNING: [generic] Falling back on generic information extractor.
[generic] kaptan-pengu-ve-arkadaslari-1: Downloading webpage
[generic] kaptan-pengu-ve-arkadaslari-1: Extracting information
[debug] Looking for video embeds
[debug] Identified a twitter:player iframe
[debug] [generic] Extracting URL: https://trtcanlitv-lh.akamaihd.net/i/TRTCOCUK_1@181844/master.m3u8
[generic] master: Requesting header
WARNING: [generic] Could not send HEAD request to https://trtcanlitv-lh.akamaihd.net/i/TRTCOCUK_1@181844/master.m3u8: HTTP Error 400: Bad Request
[generic] master: Downloading webpage
ERROR: [generic] Unable to download webpage: HTTP Error 400: Bad Request (caused by <HTTPError 400: 'Bad Request'>); please report this issue on  https://github.com/yt-dlp/yt-dlp , filling out the "Broken site" issue template properly. Confirm you are on the latest version using -U (caused by <HTTPError 400: 'Bad Request'>); please report this issue on  https://github.com/yt-dlp/yt-dlp , filling out the "Broken site" issue template properly. Confirm you are on the latest version using -U
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/common.py", line 730, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 3558, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/usr/lib/python3.9/urllib/request.py", line 523, in open
    response = meth(req, response)
  File "/usr/lib/python3.9/urllib/request.py", line 632, in http_response
    response = self.parent.error(
  File "/usr/lib/python3.9/urllib/request.py", line 561, in error
    return self._call_chain(*args)
  File "/usr/lib/python3.9/urllib/request.py", line 494, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.9/urllib/request.py", line 641, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
```

### PR DESCRIPTION
**Pull request by HobbyistDev**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

</details>

This PR add site support for trtcocuk.net.tr. This site is using nuxtjs

Fixes #2635

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [x] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### CODE REVIEW

File: yt_dlp/extractor/trtcocuk.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            'title': 'Şef Roka\'nın Lezzet Dünyası 17 Bölüm İzle TRT Çocuk',
+        }
+    }]
+
+    def _real_extract(self, url):
+        display_id = self._match_id(url)
+        webpage = self._download_webpage(url, display_id)
+        nuxtjs_data = self._search_nuxt_data(webpage, display_id)['data']
+        video_url = self._search_regex(
+            r'[\'"]?(?P<url>[^"]+)', unquote(nuxtjs_data['video']), 'video_url')
```
Comment (pukkandan) [inline yt_dlp/extractor/trtcocuk.py:36]: Neither of the tests use this code. Can you give an example?

Comment (HobbyistDev) [reply yt_dlp/extractor/trtcocuk.py:36]: Sorry, i don't get it, the `video_url` is used for `extract_m3u8_formats_and_subtitle`.

Comment (HobbyistDev) [reply yt_dlp/extractor/trtcocuk.py:36]: Should i change to something like `manifest_url` ?

Comment (pukkandan) [reply yt_dlp/extractor/trtcocuk.py:36]: sorry, the comment somehow ended up on the wrong line of code. I meant the `.replace` in L38 is not needed for either tests
```py
str(video_url).replace(r'\u002F', '/')
```

Comment (pukkandan) [reply yt_dlp/extractor/trtcocuk.py:36]: What I mean is, this works for both tests
```py
        video_url = unquote(nuxtjs_data['video'])
        formats, subtitles = self._extract_m3u8_formats_and_subtitles(video_url, display_id)
```

Comment (HobbyistDev) [reply yt_dlp/extractor/trtcocuk.py:36]: I still get error for the second test after applying your suggestion

Log:
```shell
> python test/test_download.py TestDownload.test_TrtCocukVideo_1  
[debug] Loaded 1674 extractors 
[debug] [TrtCocukVideo] Extracting URL: https://www.trtcocuk.net.tr/video/sef-rokanin-lezzet-dunyasi-17 
[TrtCocukVideo] sef-rokanin-lezzet-dunyasi-17: Downloading webpage 
[TrtCocukVideo] sef-rokanin-lezzet-dunyasi-17: Downloading m3u8 information 
ERROR: [TrtCocukVideo] sef-rokanin-lezzet-dunyasi-17: Failed to download m3u8 information: <urlopen error unknown url type: %22https> (caused by URLError('unknown url type: %22https'))
  File "[REDACTED]\extractor\common.py", line 672, in extract
    ie_result = self._real_extract(url)
  File "[REDACTED]\extractor\trtcocuk.py", line 34, in _real_extract
    formats, subtitles = self._extract_m3u8_formats_and_subtitles(
  File "[REDACTED]\extractor\common.py", line 2167, in _extract_m3u8_formats_and_subtitles
    res = self._download_webpage_handle(
  File "[REDACTED]\extractor\common.py", line 866, in _download_webpage_handle
    urlh = self._request_webpage(url_or_request, video_id, note, errnote, fatal, data=data, headers=headers, query=query, expected_status=expected_status)
  File "[REDACTED]\extractor\common.py", line 823, in _request_webpage
    raise ExtractorError(errmsg, cause=err)

  File "[REDACTED]\extractor\common.py", line 805, in _request_webpage
    return self._downloader.urlopen(self._create_request(url_or_request, data, headers, query))
  File "[REDACTED]\YoutubeDL.py", line 3680, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "[REDACTED]\lib\urllib\request.py", line 519, in open
    response = self._open(req, data)
  File "[REDACTED]\lib\urllib\request.py", line 541, in _open
    return self._call_chain(self.handle_open, 'unknown',
  File "[REDACTED]\lib\urllib\request.py", line 496, in _call_chain
    result = func(*args)
  File "[REDACTED]\lib\urllib\request.py", line 1419, in unknown_open
    raise URLError('unknown url type: %s' % type)
urllib.error.URLError: <urlopen error unknown url type: %22https>

Retrying: 1 failed tries

##########


[debug] [TrtCocukVideo] Extracting URL: https://www.trtcocuk.net.tr/video/sef-rokanin-lezzet-dunyasi-17
[TrtCocukVideo] sef-rokanin-lezzet-dunyasi-17: Downloading webpage
[TrtCocukVideo] sef-rokanin-lezzet-dunyasi-17: Downloading m3u8 information 
ERROR: [TrtCocukVideo] sef-rokanin-lezzet-dunyasi-17: Failed to download m3u8 information: <urlopen error unknown url type: %22https> (caused by URLError('unknown url type: %22https'))
  File "[REDACTED]\extractor\common.py", line 672, in extract
    ie_result = self._real_extract(url)
  File "[REDACTED]\extractor\trtcocuk.py", line 34, in _real_extract
    formats, subtitles = self._extract_m3u8_formats_and_subtitles(
  File "[REDACTED]\extractor\common.py", line 2167, in _extract_m3u8_formats_and_subtitles
    res = self._download_webpage_handle(
  File "[REDACTED]\extractor\common.py", line 866, in _download_webpage_handle
    urlh = self._request_webpage(url_or_request, video_id, note, errnote, fatal, data=data, headers=headers, query=query, expected_status=expected_status)
  File "[REDACTED]\extractor\common.py", line 823, in _request_webpage
    raise ExtractorError(errmsg, cause=err)

  File "[REDACTED]\extractor\common.py", line 805, in _request_webpage
    return self._downloader.urlopen(self._create_request(url_or_request, data, headers, query))
  File "[REDACTED]\YoutubeDL.py", line 3680, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "[REDACTED]\lib\urllib\request.py", line 519, in open
    response = self._open(req, data)
  File "[REDACTED]\lib\urllib\request.py", line 541, in _open
    return self._call_chain(self.handle_open, 'unknown',
  File "[REDACTED]\lib\urllib\request.py", line 496, in _call_chain
    result = func(*args)
  File "[REDACTED]\lib\urllib\request.py", line 1419, in unknown_open
    raise URLError('unknown url type: %s' % type)
urllib.error.URLError: <urlopen error unknown url type: %22https>

Retrying: 2 failed tries

##########


[debug] [TrtCocukVideo] Extracting URL: https://www.trtcocuk.net.tr/video/sef-rokanin-lezzet-dunyasi-17
[TrtCocukVideo] sef-rokanin-lezzet-dunyasi-17: Downloading webpage
[TrtCocukVideo] sef-rokanin-lezzet-dunyasi-17: Downloading m3u8 information 
ERROR: [TrtCocukVideo] sef-rokanin-lezzet-dunyasi-17: Failed to download m3u8 information: <urlopen error unknown url type: %22https> (caused by URLError('unknown url type: %22https'))
  File "[REDACTED]\extractor\common.py", line 672, in extract
    ie_result = self._real_extract(url)
  File "[REDACTED]\extractor\trtcocuk.py", line 34, in _real_extract
    formats, subtitles = self._extract_m3u8_formats_and_subtitles(
  File "[REDACTED]\extractor\common.py", line 2167, in _extract_m3u8_formats_and_subtitles
    res = self._download_webpage_handle(
  File "[REDACTED]\extractor\common.py", line 866, in _download_webpage_handle
    urlh = self._request_webpage(url_or_request, video_id, note, errnote, fatal, data=data, headers=headers, query=query, expected_status=expected_status)
  File "[REDACTED]\extractor\common.py", line 823, in _request_webpage
    raise ExtractorError(errmsg, cause=err)

  File "[REDACTED]\extractor\common.py", line 805, in _request_webpage
    return self._downloader.urlopen(self._create_request(url_or_request, data, headers, query))
  File "[REDACTED]\YoutubeDL.py", line 3680, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "[REDACTED]\lib\urllib\request.py", line 519, in open
    response = self._open(req, data)
  File "[REDACTED]\lib\urllib\request.py", line 541, in _open
    return self._call_chain(self.handle_open, 'unknown',
  File "[REDACTED]\lib\urllib\request.py", line 496, in _call_chain
    result = func(*args)
  File "[REDACTED]\lib\urllib\request.py", line 1419, in unknown_open
    raise URLError('unknown url type: %s' % type)
urllib.error.URLError: <urlopen error unknown url type: %22https>

WARNING: test_TrtCocukVideo_1 failed due to network errors, skipping...
.
----------------------------------------------------------------------
Ran 1 test in 6.902s
```

Comment (pukkandan) [reply yt_dlp/extractor/trtcocuk.py:36]: oh! Since the test got skipped, pytest was listing it as passed 😓

**Pull request merged**